frame/pov-limit/src/lib.rs

```diff
@@ -0,0 +1,244 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Pov Limit Pallet
+//!
+//! Pallet that controls compute usage and PoV size usage. The parameters for
+//! the usage are configured by root and are stored as `StorageValue`s.
+//!
+//! NOTE: This is only meant to be used for testing.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+pub mod weights;
+
+use blake2::{Blake2b512, Digest};
+use frame_support::{defensive, pallet_prelude::*, weights::WeightMeter};
+use frame_system::pallet_prelude::*;
+use sp_runtime::{traits::Zero, Perbill, Saturating};
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+/// Interface for wasting `ref_time`. The `waste_ref_time` function should be
+/// computation heavy.
+pub trait RefTimeWaster {
+	/// Wastes some `ref_time`. Receives a counter as an argument.
+	fn waste_ref_time(counter: u32);
+}
+
+/// Interface for wasting `proof_size`. The `waste_proof_size` should increase
+/// the size of the PoV block.
+pub trait PovWaster {
+	/// Wastes some `proof_size`. Receives a counter as an argument.
+	fn waste_proof_size(counter: u32) -> Weight;
+}
```

Comment:
Not sure if we really need to make this entire functionality generic? 

If we need to keep it generic, I would only create one trait. The functionality can be covered with one trait.

Reply:
Yea originally I thought people want to customize both independently, but in fact they probably want to customize neither…

Reply:
Initially, it made sense for the `PovWaster` since we did not have `TrashData` so each implementation would specify from where to read. But there isn't a point to have this anymore so I will make this non-generic.